src/Common/src/Interop/Windows/mincore/Interop.GetNamedPipeInfo.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Runtime.InteropServices;
+
+internal partial class Interop
+{
+    internal partial class mincore
+    {
+        [DllImport(Libraries.Kernel32, SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        internal static extern bool GetNamedPipeInfo(
```

Comment:
Looking at https://msdn.microsoft.com/en-us/library/windows/desktop/aa365445(v=vs.85).aspx  and the usage of this API in our code, it seems we are using IntPtr when we want to pass NULL to the native side( the param is _out_opt LPDWORD). So using 'out int' will have a marshaling overhead. We could consider using IntPtr instead.